main_test.py

```diff
@@ -5,58 +5,226 @@
+                { "room": None, "name": "Free Period", "teacher": None, "teacher_username": None, "department": None, "period": "H" }
+            ],
+            [
+                { "room": None, "name": "Free Period", "teacher": None, "teacher_username": None, "department": None, "period": "A" },
+                { "period": "B", "room": "TMAC-007", "name": "Evolution of Society: Economics & Technology", "teacher_username": "mdelaney", "department": "Interdisciplinary Electives" },
+                { "room": None, "name": "Free Period", "teacher": None, "teacher_username": None, "department": None, "period": "C" },
+                { "room": None, "name": "Free Period", "teacher": None, "teacher_username": None, "department": None, "period": "D" },
+                { "room": None, "name": "Free Period", "teacher": None, "teacher_username": None, "department": None, "period": "E" },
+                { "room": None, "name": "Free Period", "teacher": None, "teacher_username": None, "department": None, "period": "F" },
+                { "room": None, "name": "Free Period", "teacher": None, "teacher_username": None, "department": None, "period": "G" },
+                { "room": None, "name": "Free Period", "teacher": None, "teacher_username": None, "department": None, "period": "H" }
+            ],
+            [
+                { "room": None, "name": "Free Period", "teacher": None, "teacher_username": None, "department": None, "period": "A" },
+                { "room": None, "name": "Free Period", "teacher": None, "teacher_username": None, "department": None, "period": "B" },
+                { "room": None, "name": "Free Period", "teacher": None, "teacher_username": None, "department": None, "period": "C" },
+                { "room": None, "name": "Free Period", "teacher": None, "teacher_username": None, "department": None, "period": "D" },
+                { "room": None, "name": "Free Period", "teacher": None, "teacher_username": None, "department": None, "period": "E" },
+                { "room": None, "name": "Free Period", "teacher": None, "teacher_username": None, "department": None, "period": "F" },
+                { "room": None, "name": "Free Period", "teacher": None, "teacher_username": None, "department": None, "period": "G" },
+                { "room": None, "name": "Free Period", "teacher": None, "teacher_username": None, "department": None, "period": "H" }
+            ]
+        ],
+        "sid": 106,
+        "nickname": None,
+        "firstname": "Jonathan",
+        "lastname": "Briggs",
+        "gradyear": None,
+        "username": "jbriggs",
+        "advisor": None,
+        "grade": None
+    },
+}
+
+
+TEST_MASTER_SCHEDULE = [{
+    "2020-09-09": "Remote A-D_Rem",
+}]
+
+
+TEST_TEACHER = "jbriggs"
+TEST_STUDENT = "aaardvark"
+TEST_STUDENT_NO_PIC = "bbison"
+
+
+class FakeKey:

```

Comment:
Why are we making our own `Fake` objects? Python 3.3+ has built-in support for `Mock` and MagicMock`, and using these would be much more Pythonic.